app/src/main/java/ch/sdp/vibester/activity/BuzzerScreenActivity.kt

```diff
@@ -151,16 +266,24 @@ class BuzzerScreenActivity : AppCompatActivity() {
                 val view = map[pressedBuzzer]?.let { it1 -> findViewById<TextView>(it1) }
                 if (view != null && updater.getMap().keys.contains(pressedBuzzer)) {view.text=updater.getMap()[pressedBuzzer].toString()}
             }
-
+            if (gameManager.playingMediaPlayer()) {
+                gameManager.stopMediaPlayer()
+            }
+            setPressed(NO_BUZZER_PRESSED) // reset the buzzer
+            gameManager.setNextSong()
+            startRound(ctx, gameManager)
         }
-        setPressed(NO_BUZZER_PRESSED) // reset the buzzer
+
     }
 
 
     /**
      * Fires an intent from the Gamescreen to the Ending Screen
      */
     fun switchToEnding(view: View) {
+        if (gameManager.playingMediaPlayer()) {

```

Comment:
I had a bug that the game suddenly switched to end without stopping the media player. I am not sure how to reproduce the error, i have been playing a lot.

Reply:
Maybe it happened when the super.enRound is used and it goes to super.switvhToEnding